lib/src/license_detection/license.dart

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+import 'package:pana/src/license_detection/tokenizer.dart';
+
+@sealed
+class License {
+  /// SPDX identifier of the license, is empty in case of unknown license.
+  final String identifier;
+
+  /// Original text from the license file.
+  final String content;
+
+  /// Normalized [Token]s created from the original text.
+  final List<Token> tokens;
+
+  /// Map from [Token.value] to their number of occurrences in this license.
+  final Map<String, int> occurrences;
+
+  License._(this.content, this.tokens, this.occurrences, this.identifier);
+
+  factory License.parse(String identifier, String content) {
+    final tokens = tokenize(content);
+    final table = generateFrequencyTable(tokens);
+    return License._(content, tokens, table, identifier);
+  }
+}
+
+/// Contains details regarding the results of corpus license match with unknown text.
+@sealed
+class LicenseMatch {
+  /// Sequence of tokens from input text that were considered a match for the detected [License].
+  final List<Token> tokens;
+
+  /// Confidence score of the detected license.
+  final double confidence;
+
+  /// Detected license the input have been found to match with given [confidence].
+  final License license;
+
+  LicenseMatch(this.tokens, this.confidence, this.license);
+}
+
+/// Generates a frequency table for the given list of [tokens].
+///
+/// [Token.value] is mapped to the number of occurrences in the license text.
+@visibleForTesting
+Map<String, int> generateFrequencyTable(List<Token> tokens) {
+  var table = <String, int>{};
+
+  for (var token in tokens) {
+    table[token.value] = table.putIfAbsent(token.value, () => 0) + 1;
+  }
+
+  return table;
+}
+
+/// Load a list of [License] from the license files found in [directories].
+///
+/// The files in [directories] should be plain `.txt` and `utf-8` encoded.
+/// In case it is not `utf-8` encoded the file will be skipped.
+/// Name of the license file is expected to be in the form of `spdx-identifier.txt`.
+/// The [directories] are not searched recursively.
+/// Throws [FormatException] if any of the directories contains a file that is not a valid license file.
+/// Throws [ArgumentError] in case of a invalid [SPDX identifier][1].
```

Comment:
We don't document what Errors are thrown, we do document which exceptions are thrown.

For errors we generally just write:
> Throws if files in [directories] are not UTF-8 with names on the form `<spdx-identifier>.txt`